### PR TITLE
[virsh.guestinfo]Add the case for the interface option

### PR DESCRIPTION
**Pull request by iccaszhulili**
Signed-off-by: Lili Zhu <lizhu@redhat.com>

Test Case: RHEL-287990

JOB ID     : 688b85a3ddac9b3568da8d33b433ffdcada30bcc
JOB LOG    : /root/avocado/job-results/job-2022-03-01T05.15-688b85a/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virsh.guestinfo.positive.interface_info: PASS (64.66 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 66.11 s

### CODE REVIEW

File: libvirt/tests/src/virsh_cmd/domain/virsh_guestinfo.py [hunk 1]
```diff
@@ -252,6 +252,42 @@ def check_guest_filesystem_info(info_from_agent_cmd):
             session.close()
         return fs_info
 
+    def check_guest_interface_info():
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_guestinfo.py:255]: it sounds like there is no anywhere this method is called

Comment (iccaszhulili) [reply libvirt/tests/src/virsh_cmd/domain/virsh_guestinfo.py:255]: Hi, Line 355 called this function

                func_name = "check_guest_%s_info" % option[2:]
                info_from_guest = locals()[func_name]()

Thanks

File: libvirt/tests/src/virsh_cmd/domain/virsh_guestinfo.py [hunk 2]
```diff
@@ -252,6 +252,42 @@ def check_guest_filesystem_info(info_from_agent_cmd):
...
+        if_info = {}
+        count = -1
+        addr_count = 0
+        session = vm.wait_for_login()
+        try:
+            ip_cmd = 'ip a'
+            ip_lines = session.cmd_output(ip_cmd).strip().splitlines()
+            for line in ip_lines:
```
Comment (chunfuwen) [inline libvirt/tests/src/virsh_cmd/domain/virsh_guestinfo.py:263]: line 263 to 285 is too large, perhaps you can wrap them into independent methods

Comment (iccaszhulili) [reply libvirt/tests/src/virsh_cmd/domain/virsh_guestinfo.py]: Updated, thanks

File: libvirt/tests/src/virsh_cmd/domain/virsh_guestinfo.py [hunk 3]
```diff
@@ -252,6 +252,58 @@ def check_guest_filesystem_info(info_from_agent_cmd):
             session.close()
         return fs_info
 
+    def parse_interface_info(lines):
+        """
+        Parse the info returned from 'ip a' cmd
+
+        :param lines: The list of interface info to be parsed
+        :return: a dict of  guest interface name, mac addr, and other info
```
Comment (dzhengfy) [inline libvirt/tests/src/virsh_cmd/domain/virsh_guestinfo.py:260]: Please give an example for the returned dict for better understanding

File: libvirt/tests/src/virsh_cmd/domain/virsh_guestinfo.py [hunk 4]
```diff
@@ -252,6 +252,58 @@ def check_guest_filesystem_info(info_from_agent_cmd):
...
+        return if_info
+
+
+    def check_guest_interface_info():
+        """
+        check the info of guest interfaces from guest side
+
+        :return: the guest interface info from guest side
+        """
+        session = vm.wait_for_login()
```
Comment (dzhengfy) [inline libvirt/tests/src/virsh_cmd/domain/virsh_guestinfo.py:297]: Please define 'if_info' before  try clause to avoid potential reference error.

**Review by dzhengfy — COMMENTED** [review]
Others LGTM

**Comment by dzhengfy** [comment]
@iccaszhulili please fix the CI failures.

File: libvirt/tests/src/virsh_cmd/domain/virsh_guestinfo.py (hunk 4 above)
Comment (iccaszhulili) [reply libvirt/tests/src/virsh_cmd/domain/virsh_guestinfo.py:305]: Updated, thanks

File: libvirt/tests/src/virsh_cmd/domain/virsh_guestinfo.py (hunk 3 above)
Comment (iccaszhulili) [reply libvirt/tests/src/virsh_cmd/domain/virsh_guestinfo.py]: Updated, thanks

**Review by chunfuwen — APPROVED** [review]
LGTM

**Review by dzhengfy — COMMENTED** [review]
LGTM

**Review by dzhengfy — APPROVED** [review]
LGTM

**Pull request merged**